### DISCUSSION TITLE
Update affects existing data?

### DISCUSSION THREAD

**Discussion by cubytz** [flipt-io/flipt#789] [Q&A] [answered]
Thank you for the great project! I have a question (sorry if I am missing something): Does updating (1.6.1 -> 1.7.0 or any other version for this matter) affect the database in any way (for example, erase already persisted data)? 

Thank you so much for the great library!

**Comment by markphelps** [comment, marked as answer]
Hey @cubytz , great question!

We take great care to ensure that no data is lost or corrupted between upgrades. But to fully answer your question, it really depends on how you run Flipt.

For example if you are running it via Docker and using the default SQLite db, then you must [mount the volume from the host machine](https://flipt.io/docs/installation/#run-the-image) in order for the data to persist.

If you are running Flipt 'natively' on a Linux host, then the default SQLite db will be persisted as normal and remain between upgrades since it's just a regular file on the host.

If you are running with either [MySQL or Postgres](https://flipt.io/docs/configuration/#databases) then the data is persisted in those DBs and Flipt will not erase any data between upgrades.

So basically the only way Flipt will lose data between versions is if you are running it via Docker AND using the default SQLite configuration AND not mounting the data volume, but thats really just the nature of Docker and not specific to Flipt.

Hope this helps
